snapshots/src/test/java/io/axoniq/dev/samples/rest/TestUtils.java

```diff
@@ -0,0 +1,68 @@
+package io.axoniq.dev.samples.rest;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+import org.testcontainers.containers.GenericContainer;
+
+public class TestUtils {
+
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private static final String COMMAND_HANDLERS = "http://%s:%s/v1/components/application/commands?context=default";
+    private static final RestTemplate restTemplate = new RestTemplate();
+
+    private TestUtils(){
+        // utility class
+    }
+
+    public static void waitForCommandHandlers(GenericContainer<?> axonServer, int expectedNumberOfCommandHandlers, int time, TimeUnit unit) {
```

Comment:
I understand you weren't overly happy with this as it stands.

There are, I believe, two things you could do to mitigate this:

1. Have a Spring Event nudge you when all command handlers are registered according to your Axon application. This is, as I believe you know, a pending issue for Axon Framework (as found [here](https://github.com/AxonFramework/AxonFramework/issues/880)). This makes it something you cannot do right now unless you built it yourself. As such, this solution would I wager win.
2. Not use Spring to wire everything together. It is Spring which makes it uncertain when the command handlers are registered. If instead, you would use Axon's Configuration API directly, you would be aware when you register the command handlers since you are in control. This would allow you to remove this utility entirely but does require you to change the way you are configuring your tests and sample project.